src/Swagger/ApiDocumentationBuilder.php

```diff
@@ -114,51 +113,61 @@ public function getApiDocumentation()
                 $context['serializer_groups'] = isset($context['serializer_groups']) ? array_merge($context['serializer_groups'], $attributes['denormalization_context']['groups']) : $context['serializer_groups'];
             }
 
+            $definitions[$shortName] = [
+                'type' => 'object',
+            ];
             foreach ($this->propertyNameCollectionFactory->create($resourceClass, $context) as $propertyName) {
                 $propertyMetadata = $this->propertyMetadataFactory->create($resourceClass, $propertyName);
-
                 if ($propertyMetadata->isIdentifier() && !$propertyMetadata->isWritable()) {
                     continue;
                 }
-                $range = $this->getRange($propertyMetadata);
-
-                $property[$propertyName] = [
-                    'type' => $range,
-                ];
 
-                if (is_array($range)) {
-                    $property[$propertyName] = $range;
+                if ($propertyMetadata->isRequired()) {
+                    $definitions[$shortName]['required'][] = $propertyName;
                 }
 
-                $required = [];
 
-                if ($propertyMetadata->isRequired()) {
-                    $required = array_merge($required, [$propertyName]);
+                $range = $this->getRange($propertyMetadata);
+                if (null === $range) {
+                    continue;
                 }
 
-                if (!empty($required)) {
-                    $properties[$shortName]['required'] = $required;
-                }
+                $definitions[$shortName]['properties'][$propertyName]['description'] = $propertyMetadata->getDescription() ?:  '';
```

Comment:
Is `description` mandatory? If it is optional it would be better to omit this field instead of setting it to an empty string.